includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1563,4 +1577,15 @@ private function get_taxes_like_cart( $product, $price ) {
 		// Normally there should be a single tax, but `calc_tax` returns an array, let's use it.
 		return WC_Tax::calc_tax( $price, $rates, false );
 	}
+
+	/**
+	 * Wrapper for `wc_get_shipping_method_count` to allow mocking.
+	 *
+	 * @param boolean $include_legacy Whether to include legacy shipping methods.
+	 * @param boolean $enabled_only   Whether to include only enabled shipping methods.
+	 * @return int                    The number of shipping methods.
+	 */
+	public function get_shipping_method_count( $include_legacy = false, $enabled_only = false ) {
+		return wc_get_shipping_method_count( $include_legacy, $enabled_only );
+	}
```

Comment:
There are a few approaches we can try to avoid adding some small technical debt here for testing purposes: 
1. control shipping methods from the test via `// add_filter('woocommerce_shipping_zone_shipping_methods', '__return_empty_array', PHP_INT_MAX);` or similar e.g. 
2. run the following queries in test to manage shipping methods
```php
$wpdb->query(
			$wpdb->prepare(
				"UPDATE {$wpdb->prefix}woocommerce_shipping_zone_methods
				SET is_enabled = 0
				WHERE is_enabled = 1"
			)
		);
		$wpdb->query("DELETE FROM {$wpdb->prefix}woocommerce_shipping_zone_methods");
		WC()->shipping()->shipping_methods = [];
```
3. remove shipping method from the shipping zone through `$this->zone->delete_shipping_method($id);`

I tried them all and (3) seems to work, I've added suggestion in the code below in the test class. This will allow us to get rid of `get_shipping_method_count`

Reply:
Thanks for the tip with `sleep()` here! I tried several different methods for removing shipping methods but nothing had an effect. Waiting a second does the trick!